src/parquet/encodings/encodings.h

```diff
@@ -67,6 +67,38 @@ class Decoder {
   int num_values_;
 };
 
+
+// Base class for value encoders. Since encoders may or not have state (e.g.,
+// dictionary encoding) we use a class instance to maintain any state.
+//
+// TODO(wesm): Encode interface API is temporary
+template <int TYPE>
+class Encoder {
+ public:
+  typedef typename type_traits<TYPE>::value_type T;
+
+  virtual ~Encoder() {}
+
+  // Subclasses should override the ones they support
+  //
+  // @returns: the number of bytes written to dst
+  virtual size_t Encode(const T* src, int num_values, uint8_t* dst) {
```

Comment:
No, I'm not sure, I was also thinking about this because these classes have no ability to allocate memory. Let's create a JIRA to do that before we get in too deep with this assumption; I don't think this JIRA is the right place for it though...